apis/logging/v1/filter_types.go

```diff
@@ -0,0 +1,18 @@
+package v1
+
+// NOTE: The Enum validation on FilterSpec.Type must be updated if the list of
+// known types changes.
+
+// Filter type constants, must match JSON tags of FilterTypeSpec fields.
+const (
+	FilterKubeAPIAudit = "kubeAPIAudit"
+)
+
+// FilterTypeSpec is a union of filter specification types.
+// The fields of this struct define the set of known filter types.
+type FilterTypeSpec struct {
+	// +optional
+	*KubeAPIAudit `json:"kubeAPIAudit,omitempty"`
```

Comment:
I'm not certain, but don't we need named fields here to tell apart different filter types, once we have > 1 of them?

Reply:
Nope, with "anonymous" fields like this the type name becomes the field names. Since all the options will be different types, they'll have different names. That said, there is no good reason for the un-named field here so maybe I should remove it by the law of least surprise.